appshell/appshell_extensions_mac.mm

```diff
@@ -1175,6 +1175,154 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
     return NO_ERROR;
 }
 
+OSStatus _RunToolWithAdminPrivileges( AuthorizationRef authorizationRef, const std::string& tool, const std::vector<std::string> &args)
+{
+    OSStatus status = 0;
+    if(!authorizationRef || tool.length() ==  0 || args.size() == 0)
+        return errAuthorizationInvalidSet;
+    
+    FILE *pipe = NULL;
+
+    // convert to std::vector<char *> array.
+    std::vector<char*> argv(1 + args.size(), NULL);
+
+    for (size_t i = 0; i < args.size(); ++i)
+        argv[i] = const_cast<char*>(args[i].c_str());
+    
+    // This is a deprecated API. Apple recommends a dedicated helper to
+    // fix this issue. This ideally should be replaced by SMBless API.
+    status = AuthorizationExecuteWithPrivileges(authorizationRef, tool.c_str(),
+                                                kAuthorizationFlagDefaults, &argv[0], &pipe);
+    
+    if(pipe){
+        fclose(pipe);
+    }
+    
+    return status;
+    
+}
+
+int32 InstallCommandLineTools()
+{
+    // Create authorization reference
+    OSStatus authStatus  = 0;
+    OSStatus toolStatus  = 0;
+    int32    errorCode   = NO_ERROR;
+
+
+    std::string destFile   = "/usr/local/bin/Brackets";
+    std::string destFolder = "/usr/local/bin";
+
+    std::string rmTool     = "/bin/rm";
+    std::string rmArgs     = "-f";
+
+    std::string mkDirTool  = "/bin/mkdir";
+    std::string mkDirArgs  = "-p";
+
+    std::string lnTool     = "/bin/ln";
+    std::string lnArgs     = "-s";
+
+
+    AuthorizationRef authorizationRef = NULL;
+    
+    // AuthorizationCreate and pass NULL as the initial
+    // AuthorizationRights set so that the AuthorizationRef gets created
+    // successfully.
+    // http://developer.apple.com/qa/qa2001/qa1172.html
+    
+    authStatus = AuthorizationCreate(NULL, kAuthorizationEmptyEnvironment,
```

Comment:
Why not include this as well in the try catch block? In case an exception is thrown by OS